transport/tlscommon/ca_pinning_test.go

```diff
@@ -15,6 +15,8 @@
 // specific language governing permissions and limitations
 // under the License.
 
+//go:build !requirefips
+
```

Comment:
The only thing I didn't understand is why most of the tests won't run when FIPS is enabled.
Do they break with FIPS enabled?

Reply:
Some of them break, but this is mostly so we can limit the `requirefips` tests to only test the different behaviour

Reply:
@michel-laterman I don't think that we should generally exclude non-system tests from running when in fips mode. What we had discussed was to exclude the expensive system tests that might test certain specifics that aren't focused on FIPS relevant parts; instead we want to have some focused system tests for FIPS covering TLS, crypto, pgp,.. 

Would be worth looking into why these tests are failing rather than disabling them.

Reply:
I agree with @simitt, because FIPS changes some cryptographic libraries it is better to have the full suit of tests dealing with crypto running and only add build tags to the ones with different behaviour.

Also, IIRC the `elastic-agent-libs` CI is rather fast, so I don't see any issues with running the tests twice, one for each build.